packages/core/contracts/financial-templates/expiring-multiparty/PricelessPositionManager.sol

```diff
@@ -784,18 +782,64 @@ contract PricelessPositionManager is FeePayer {
         return OracleInterface(finder.getImplementationAddress(OracleInterfaces.Oracle));
     }
 
+    function _getOptimisticOracle() internal view returns (OptimisticOracleInterface) {
+        return OptimisticOracleInterface(finder.getImplementationAddress(OracleInterfaces.OptimisticOracle));
+    }
+
     function _getFinancialContractsAdminAddress() internal view returns (address) {
         return finder.getImplementationAddress(OracleInterfaces.FinancialContractsAdmin);
     }
 
     // Requests a price for transformed `priceIdentifier` at `requestedTime` from the Oracle.
-    function _requestOraclePrice(uint256 requestedTime) internal {
+    function _requestOraclePriceExpiration(uint256 requestedTime) internal {
+        OptimisticOracleInterface optimisticOracle = _getOptimisticOracle();
+
+        // Increase token allowance to enable the optimistic oracle reward transfer.
+        FixedPoint.Unsigned memory reward = _computeFinalFees();
+        collateralCurrency.safeIncreaseAllowance(address(optimisticOracle), reward.rawValue);
+        optimisticOracle.requestPrice(
+            _transformPriceIdentifier(requestedTime),
+            requestedTime,
+            _getAncillaryData(),
+            collateralCurrency,
+            reward.rawValue // Reward is equal to the final fee
```

Comment:
Why do you set reward equal to final fee? Is this just an arbitrary default choice or is there a reason behind this.

Reply:
I think this is just an arbitrary default choice -- it also matches what the contract was paying out previously to the DVM.

Reply:
Cool SGTM. As long as no changes to economics I agree